src/Analysis/Codelyzer.Analysis/CodeAnalyzer.cs

```diff
@@ -55,6 +55,8 @@ protected CodeAnalyzer(AnalyzerConfiguration configuration, ILogger logger)
         /// <returns></returns>
         public abstract Task<List<AnalyzerResult>> AnalyzeSolution(string solutionPath, Dictionary<string, List<string>> originalReferences, Dictionary<string, List<string>> references);
 
+        public abstract CodeGraph GenerateGraph(List<AnalyzerResult> analyzerResults);
```

Comment:
nit: We might want to add documentation later for consistency

Reply:
Will add